rasa/nlu/data_router.py

```diff
@@ -348,52 +314,49 @@ def start_train_process(
             )
             self.project_store[project].status = STATUS_TRAINING
 
-        def training_callback(model_path):
-            model_dir = os.path.basename(os.path.normpath(model_path))
-            self.project_store[project].update(model_dir)
-            self._current_training_processes -= 1
-            self.project_store[project].current_training_processes -= 1
-            if (
-                self.project_store[project].status == STATUS_TRAINING
-                and self.project_store[project].current_training_processes == 0
-            ):
-                self.project_store[project].status = STATUS_READY
-            return model_path
-
-        def training_errback(failure):
-            logger.warning(failure)
-
-            self._current_training_processes -= 1
-            self.project_store[project].current_training_processes -= 1
-            self.project_store[project].status = STATUS_FAILED
-            self.project_store[project].error_message = str(failure)
-
-            return failure
+        loop = asyncio.get_event_loop()
 
         logger.debug("New training queued")
 
         self._current_training_processes += 1
         self.project_store[project].current_training_processes += 1
 
-        result = self.pool.submit(
+        task = loop.run_in_executor(
+            self.pool,
             do_train_in_worker,
             train_config,
             data_file,
-            path=self.project_dir,
-            project=project,
-            fixed_model_name=model_name,
-            storage=self.remote_storage,
```

Comment:
I personally find it actually easier to read with the key words (e.g. in Swift you always have to use keywords as soon as you pass more than one parameter). Up to you.